cloudinit/sources/DataSourceAeza.py

```diff
@@ -0,0 +1,172 @@
+                raise e
+            # Wait before retrying if not a 404 error and retries are left
+            time.sleep(sec_between_retries)
+            attempt += 1
+        except RequestException as e:
+            if attempt == retries:
+                # If it's the last attempt, re-raise the last exception
+                raise e
+            # Wait before retrying if retries are left
+            time.sleep(sec_between_retries)
+            attempt += 1
+        # If max_wait timeout is enabled check for max wait timeout
+        if max_wait_seconds is not None:
+            if start + max_wait_seconds < time.time():
+                raise RequestException(
+                    f"Failed to retrieve data from {url} after {retries + 1} attempts because of max wait timeout."
+                )
+
+    # If retries are exhausted, raise an exception indicating the failure
+    raise RequestException(f"Failed to retrieve data from {url} after {retries + 1} attempts.")
+
+
+def read_metadata(url, timeout_seconds=2, max_wait_seconds=120, sec_between_retries=2, retries=30):
+    """
+    Read user data from URL or raise error.
+    """
+    response = read_url(
+        url,
+        timeout_seconds=timeout_seconds,
+        max_wait_seconds=max_wait_seconds,
+        sec_between_retries=sec_between_retries,
+        retries=retries,
+    )
+    if not response.ok:
+        raise RuntimeError("unable to read metadata at %s" % url)
+    return util.load_json(response.content.decode())
+
+
+def read_data(url, timeout_seconds=2, max_wait_seconds=120, sec_between_retries=2, retries=30):
+    """
+    Read data from URL.
+    """
+    try:
+        response = read_url(
+            url,
+            timeout_seconds=timeout_seconds,
+            max_wait_seconds=max_wait_seconds,
+            sec_between_retries=sec_between_retries,
+            retries=retries,
+        )
+        if not response.ok:
+            raise RuntimeError("unable to read data at %s" % url)
+        return response.content
+    except HTTPError as e:
+        if e.response.status_code == 404:
+            return None
+        raise e
+
+
+def format_url(url, system_uuid=None):
+    if system_uuid is None:
+        system_uuid = dmi.read_dmi_data("system-uuid")
+    return url.format(id=system_uuid)
+
+
+class DataSourceAeza(sources.DataSource):
+
+    dsname = "Aeza"
+
+    def __init__(self, sys_cfg, distro, paths, ud_proc=None):
+        super().__init__(sys_cfg, distro, paths, ud_proc)
+
+        self.ds_cfg = util.mergemanydict([self.ds_cfg, BUILTIN_DS_CONFIG])
+
+        url_params = self.get_url_params()
+        self.timeout_seconds = url_params.timeout_seconds
+        self.max_wait_seconds = url_params.max_wait_seconds
+        self.retries = url_params.num_retries
+        self.sec_between_retries = url_params.sec_between_retries
+
+        self.metadata_address = self.ds_cfg["metadata_url"]
```

Comment:
This is good to set in __init__ and on the instance itself because [get_sub_platform](https://github.com/canonical/cloud-init/blob/main/cloudinit/sources/__init__.py#L679) references this instance atttribute for user/script-facing instance data queries like `cloud-init query v1` or `cloud-id --long`.